tests/attacks/test_knockoff_nets.py

```diff
@@ -0,0 +1,434 @@
+# MIT License
+#
+# Copyright (C) IBM Corporation 2018
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+import unittest
+
+import tensorflow as tf
+import numpy as np
+import keras.backend as k
+import keras
+from keras.models import Sequential
+from keras.layers import Dense, Flatten, Conv2D, MaxPooling2D
+
+from art.attacks.extraction.knockoff_nets import KnockoffNets
+from art.classifiers import TensorFlowClassifier
+from art.classifiers import KerasClassifier
+from art.classifiers import PyTorchClassifier
+from art.utils import load_dataset, master_seed
+from tests.utils_test import get_classifier_tf
+from tests.utils_test import get_classifier_kr
+from tests.utils_test import get_classifier_pt
+from tests.utils_test import get_iris_classifier_tf
+from tests.utils_test import get_iris_classifier_kr
+from tests.utils_test import get_iris_classifier_pt
+from art.config import ART_NUMPY_DTYPE
+
+logger = logging.getLogger(__name__)
+
+try:
+    # Conditional import of `torch` to avoid segmentation fault errors this framework generates at import
+    import torch
+    import torch.nn as nn
+    import torch.optim as optim
+except ImportError:
+    logger.info('Could not import PyTorch in utilities.')
+
+if tf.__version__[0] == '2':
+    # pylint: disable=E0401
+    import tensorflow.compat.v1 as tf
+    tf.disable_eager_execution()
```

Comment:
Please remove these lines as they are not required anymore.

Reply:
Solved.